numpy/core/defchararray.py

```diff
@@ -452,7 +451,7 @@ def center(a, width, fillchar=' '):
     width_arr = numpy.asarray(width)
     size = int(numpy.max(width_arr.flat))
     if numpy.issubdtype(a_arr.dtype, numpy.string_):
-        fillchar = asbytes(fillchar)
+        fillchar = numpy.asanyarray(fillchar, dtype=numpy.string_)
```

Comment:
I'm curious... What's the behavior with these lines removed?

Reply:
I think the answer is "mixing bytes and string starts erroring", which sounds like a good thing, given that's what `"test".center(5, b"a")` does without numpy.

Reply:
Exactly. A bit of an odd side effect of the previous & current implementations is that `fillchar` can be any object as long as `len(bytes(fillchar)) == 1` and `a` consists of `bytes`.

Reply:
I suppose we could also go the `fillchar  = np.asanyarray(fillchar, dtype=a_arr.dtype)` route, thus extending this (unintended?) behavior to strings.

Reply:
I'd argue the behavior is not just unintended, it's unwanted.

Reply:
I would agree with you, yes.
The only thing which is somewhat problematic is that this piece of code has been in place for over a decade (see https://github.com/numpy/numpy/commit/d820db296b9d6b281dd6f79c666e1596141b9b4a) and I fear that flat out removing it will break something somewhere.

Reply:
Still, issuing a `DeprecationWarning` might not be a bad idea (though let's safe that for a follow up pull request).